app/components/steps_component.rb

```diff
@@ -0,0 +1,25 @@
+class StepsComponent < GovukComponent::Base
+  attr_reader :title
+
+  def initialize(title: nil, classes: [], html_attributes: {})
+    super(classes: classes, html_attributes: html_attributes)
+
+    @title = title
+  end
+
+  renders_many :steps, lambda { |label:, current: false, completed: false|
+    tag.li(class: "steps-component__step#{step_class(current, completed)}") { tag.h3(class: "govuk-heading-s") { label } }
+  }
+
+  def step_class(current, completed)
```

Comment:
```ruby
def step_class(current, completed)
  return "steps-component__step steps-component__step--current" if current
  return "steps-component__step steps-component__step--completed" if completed
  
  "steps-component__step"  
end
```

Depending on the way you've implemented your styling, you might not need to include the `steps-component__step` class too for the current/completed steps

Reply:
Did you try doing?

```ruby
def step_class(current, completed)
  return "steps-component__step--current" if current
  return "steps-component__step--completed" if completed
  
  "steps-component__step"
end
```